internal/pkg/githubapi/github.go

```diff
@@ -106,136 +106,165 @@ func (ghPrClientDetails *GhPrClientDetails) getBlameURLPrefix() string {
+	case "show-plan":
+		err = handleShowPlanPREvent(ctx, ghPrClientDetails, eventPayload)
+	}
+
+	if err != nil {
+		ghPrClientDetails.PrLogger.Errorf("Handling of PR event failed: err=%s\n", err)
+	}
+}
 
+// eventToHandle returns the event to be handled, translated from a Github
+// world into the Telefonistka world. If no event should be handled, ok is
+// false.
+func eventToHandle(eventPayload *github.PullRequestEvent) (event string, ok bool) {
+	switch {
+	case *eventPayload.Action == "closed" && *eventPayload.PullRequest.Merged:
+		return "merged", true
+	case *eventPayload.Action == "opened" || *eventPayload.Action == "reopened" || *eventPayload.Action == "synchronize":
+		return "changed", true
+	case *eventPayload.Action == "labeled" && DoesPrHasLabel(eventPayload.PullRequest.Labels, "show-plan"):
+		return "show-plan", true
+	default:
+		return "", false
+	}
+}
+
+func handleShowPlanPREvent(ctx context.Context, ghPrClientDetails GhPrClientDetails, eventPayload *github.PullRequestEvent) error {
+	ghPrClientDetails.PrLogger.Infoln("Found show-plan label, posting plan")
 	defaultBranch, _ := ghPrClientDetails.GetDefaultBranch()
 	config, err := GetInRepoConfig(ghPrClientDetails, defaultBranch)
 	if err != nil {
 		ghPrClientDetails.PrLogger.Infof("Couldn't get Telefonistka in-repo configuration: %v", err)
 	}
+	promotions, _ := GeneratePromotionPlan(ghPrClientDetails, config, *eventPayload.PullRequest.Head.Ref)
+	commentPlanInPR(ghPrClientDetails, promotions)
+	return nil
+}
 
-	if *eventPayload.Action == "closed" && *eventPayload.PullRequest.Merged {
-		SetCommitStatus(ghPrClientDetails, "pending")
-		wasCommitStatusSet = true
-		err := handleMergedPrEvent(ghPrClientDetails, approverGithubClientPair.v3Client)
-		if err != nil {
-			prHandleError = err
-			ghPrClientDetails.PrLogger.Errorf("Handling of merged PR failed: err=%s\n", err)
-		}
-	} else if *eventPayload.Action == "opened" || *eventPayload.Action == "reopened" || *eventPayload.Action == "synchronize" {
-		SetCommitStatus(ghPrClientDetails, "pending")
-		wasCommitStatusSet = true
-		botIdentity, _ := GetBotGhIdentity(mainGithubClientPair.v4Client, ctx)
-		err = MimizeStalePrComments(ghPrClientDetails, mainGithubClientPair.v4Client, botIdentity)
+func handleChangedPREvent(ctx context.Context, mainGithubClientPair GhClientPair, ghPrClientDetails GhPrClientDetails, eventPayload *github.PullRequestEvent) error {
+	botIdentity, _ := GetBotGhIdentity(mainGithubClientPair.v4Client, ctx)
+	err := MimizeStalePrComments(ghPrClientDetails, mainGithubClientPair.v4Client, botIdentity)
+	if err != nil {
+		return fmt.Errorf("minimizing stale PR comments: %w", err)
+	}
+	defaultBranch, _ := ghPrClientDetails.GetDefaultBranch()
+	config, err := GetInRepoConfig(ghPrClientDetails, defaultBranch)
+	if err != nil {
+		ghPrClientDetails.PrLogger.Infof("Couldn't get Telefonistka in-repo configuration: %v", err)
+	}
```

Comment:
Same as above - getting the repo config is must for what comes next, not point in continuing.
*if* you fix it above, you can also fix it here